foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/databaseaccess/DatabasePlatform.java

```diff
@@ -3767,4 +3774,47 @@ public boolean checkTableExists(final DatabaseSessionImpl session,
         }
     }
 
+    // Common JSON types support:
+    // Stores JsonValue instances as VARCHAR.
+    /**
+     * INTERNAL:
+     * Convert JSON value field to classification type.
+     *
+     * @param <T> classification type
+     * @param jsonValue source JSON value field
+     * @return converted classification type
+     */
+    @SuppressWarnings("unchecked")
+    public <T> T convertJsonValueToDataValue(final JsonValue jsonValue) {
+        final StringWriter sw = new StringWriter(128);
+        try (final JsonWriter jw = Json.createWriter(sw)) {
+            jw.write(jsonValue);
+        }
+        return (T) sw.toString();
+    }
+
+    /**
+     * Convert classification type to JSON value field.
+     *
+     * @param jdbcValue source classification type value from JDBC
+     * @return converted JSON field value
+     */
+    public JsonValue convertDataValueToJsonValue(Object jdbcValue) {
+        try (final JsonReader jr = Json.createReader(new StringReader((String)jdbcValue))) {
+            return jr.readValue();
+        }
+    }
+
+    /**
+     * Retrieve JSON data from JDBC ResultSet.
+     *
+     * @param resultSet source JDBC ResultSet
+     * @param columnNumber index of column in JDBC ResultSet
+     * @return JSON data from JDBC ResultSet
+     * @throws SQLException if data could not be retrieved
+     */
+    public Object getJsonDataFromResultSet(ResultSet resultSet, int columnNumber) throws SQLException {
```

Comment:
I do not think these methods belongs here. Isn't there better/different way? How is the same problem being solved for other datatypes from external libraries like Oracle specific datatypes?

Reply:
Well, this has to be in platform, see this in https://docs.oracle.com/en/database/oracle/oracle-database/21/jajdb/oracle/sql/json/package-summary.html
```
OracleJsonObject pear = rs.getObject(1, OracleJsonObject.class);
```
I have to use Oracle 21c specific call using platform specific data type (`oracle.sql.json.OracleJsonValue`). This must be in Platform clas, not in common EclipseLink code.

Reply:
Also Postgres10+ platform has platform specific code to read JSON from ResultSet.